pkg/apis/admission/types.go

```diff
@@ -56,6 +56,27 @@ type AdmissionRequest struct {
 	// "binding", and kind "Binding".
 	// +optional
 	SubResource string
+
+	// RequestKind is the type of object being manipulated by the the original API request.  For example: Pod
+	// If this differs from the value in "kind", an equivalent match and conversion was performed.
+	// See documentation for the "matchPolicy" field in the webhook configuration type.
+	// +optional
+	RequestKind *metav1.GroupVersionKind
```

Comment:
it's unclear what this is when there was not conversion happening? Will it be nil? Or the equal value? Same below.

Reply:
we will always populate it, but old apiservers will not (obviously), so someone writing a webhook using our go types to work against pre-1.15 apiservers can not assume they'll get a value here. we can strengthen the spec in v1 admissionreview